frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1671,6 +1687,134 @@ class SegmentsView extends React.Component<Props, State> {
+          onRenameEnd={this.onRenameEnd}
+          getMultiSelectMenu={this.getMultiSelectMenu}
+          activeVolumeTracing={this.props.activeVolumeTracing}
+        />
+      );
+    } else {
+      const { id, name } = treeItem;
+      const isEditingDisabled = !this.props.allowUpdate;
+      const onOpenContextMenu = (event: React.MouseEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        const getMenu = (): MenuProps => ({
+          items: _.flatten([
+            {
+              key: "create",
+              onClick: () => {
+                this.createGroup(id);
+                this.hideContextMenu();
+              },
+              disabled: isEditingDisabled,
+              icon: <PlusOutlined />,
+              label: "Create new group",
+            },
+            {
+              key: "delete",
+              disabled: isEditingDisabled,
+              onClick: () => {
+                this.handleDeleteGroup(id);
+                this.hideContextMenu();
+              },
+              icon: <DeleteOutlined />,
+              label: "Delete group",
+            },
+            this.getExpandSubgroupsItem(id),
+            this.getCollapseSubgroupsItem(id),
+            this.getMoveSegmentsHereMenuItem(id),
+            {
+              key: "groupAndMeshActionDivider",
+              label: <Divider style={{ marginBottom: 0, marginTop: 0 }} />,
+              disabled: true,
+            },
+            this.getSetGroupColorMenuItem(id),
+            this.getShowSegmentStatistics(id),
+            this.getLoadMeshesFromFileMenuItem(id),
+            this.getComputeMeshesAdHocMenuItem(id),
+            this.getReloadMenuItem(id),
+            this.getRemoveMeshesMenuItem(id),
+            this.maybeGetShowOrHideMeshesMenuItems(id),
+            this.getDownLoadMeshesMenuItem(id),
+          ]),
+        });
+
+        const [x, y] = getContextMenuPositionFromEvent(event, "segment-list-context-menu-overlay");
+        this.showContextMenuAt(x, y, getMenu());
+      };
+
+      // Make sure the displayed name is not empty
+      const displayableName = name?.trim() || "<Unnamed Group>";
+
+      return (
+        <div onContextMenu={onOpenContextMenu}>
+          <FolderOutlined className="icon-margin-right" />
+          <EditableTextLabel
+            value={displayableName}
+            label="Group Name"
+            onChange={(name) => {
+              if (this.props.visibleSegmentationLayer != null) {
+                api.tracing.renameSegmentGroup(id, name, this.props.visibleSegmentationLayer.name);
+              }
+            }}
+            margin={0}
+            // The root group must not be removed or renamed
+            disableEditing={!this.props.allowUpdate || id === MISSING_GROUP_ID}
+            onRenameStart={this.onRenameStart}
+            onRenameEnd={this.onRenameEnd}
+          />
+          {this.getSegmentStatisticsModal(id)}
+        </div>
+      );
+    }
+  };
+
   render() {
```

Comment:
the git diff is a bit messed up. I simply inserted `onCheck` below the `render()` method.